plugins/CoreUpdater/Updater.php

```diff
@@ -353,7 +354,7 @@ public function getArchiveUrl($version, $https = true)
         $channel = $this->releaseChannels->getActiveReleaseChannel();
         $url = $channel->getDownloadUrlWithoutScheme($version);
 
-        if (Http::isUpdatingOverHttps() && $https) {
+        if (Http::isUpdatingOverHttps() && $https && GeneralConfig::getConfigValue('force_matomo_http_request') === 0) {
```

Comment:
This might not work. You can't rely on specific types for config values. In most cases a string value is returned (even if it's an integer value). So either you need to case the value to integer, or use `==`